build-support/bin/generate_github_workflows.py

```diff
@@ -403,16 +381,22 @@ def rust_caches(self) -> Sequence[Step]:
                 "uses": "actions/cache@v3",
                 "with": {
                     "path": f"~/.rustup/toolchains/{rust_channel()}-*\n~/.rustup/update-hashes\n~/.rustup/settings.toml\n",
-                    "key": f"{self.platform_name()}-rustup-{hashFiles('rust-toolchain')}-v1",
+                    "key": f"{self.platform_name()}-rustup-{hashFiles('rust-toolchain')}-v2",
                 },
             },
             {
                 "name": "Cache Cargo",
-                "uses": "actions/cache@v3",
+                "uses": "benjyw/rust-cache@461b9f8eee66b575bce78977bf649b8b7a8d53f1",
```

Comment:
Should probably fork this to the `pantsbuild` org and tag it.

Reply:
My hope is to upstream it, but that is the fallback.